File: src/main/java/com/example/onionstore/domain/refund/entity/RefundStatus.java
package com.example.onionstore.domain.refund.entity;

public enum RefundStatus {

    PENDING_APPROVAL {
        @Override
        public boolean canTransitTo(RefundStatus target) {
            return target == REQUESTED || target == REJECTED;
        }
    },

    REQUESTED {
        @Override
        public boolean canTransitTo(RefundStatus target) {
            return target == COMPLETED || target == FAILED;
        }
    },

    COMPLETED {
        @Override
        public boolean canTransitTo(RefundStatus target) {
            return false;
        }
    },

    REJECTED {
        @Override
        public boolean canTransitTo(RefundStatus target) {
            return false;
        }
    },

    FAILED {
        @Override
        public boolean canTransitTo(RefundStatus target) {
            return false;
        }
    };

    public abstract boolean canTransitTo(RefundStatus target);
}

File: frontend/src/constants/refundStatus.ts
import type { RefundStatus } from '@/types/refund'

export const REFUND_STATUS_LABEL: Record<RefundStatus, string> = {
  PENDING_APPROVAL: '승인 대기',
  REQUESTED: '취소 처리 중',
  COMPLETED: '환불 완료',
  REJECTED: '환불 거절',
  FAILED: '환불 실패',
}
